msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java

```diff
@@ -1108,22 +1110,22 @@ public PrivateTopic createPrivateForumTopic(String title, boolean forumIsParent,
      * Save a private forum topic
      */
 
-    public void savePrivateForumTopic(PrivateTopic topic){
-    	savePrivateForumTopic(topic, getCurrentUser());
+    public PrivateTopic savePrivateForumTopic(PrivateTopic topic){
+    	return savePrivateForumTopic(topic, getCurrentUser());
     }
 
-    public void savePrivateForumTopic(PrivateTopic topic, String userId) {
-    	savePrivateForumTopic(topic, userId, getContextId());
+    public PrivateTopic savePrivateForumTopic(PrivateTopic topic, String userId) {
+    	return savePrivateForumTopic(topic, userId, getContextId());
     }
 
-    public void savePrivateForumTopic(PrivateTopic topic, String userId, String siteId) {
+    public PrivateTopic savePrivateForumTopic(PrivateTopic topic, String userId, String siteId) {
     	boolean isNew = topic.getId() == null;
 
     	topic.setModified(new Date());
     	if(userId != null){
     		topic.setModifiedBy(userId);
     	}
-    	getHibernateTemplate().saveOrUpdate(topic);
+    	topic = getHibernateTemplate().merge(topic);
```

Comment:
I think it is good idea to treat topic as final and simply create a new var called "mergedTopic", it is clearer about what happened.

Reply:
Solved